src/main/webapp/app/overview/course-statistics/course-statistics.component.html

```diff
@@ -80,34 +73,83 @@ <h2 class="text-center">{{ 'artemisApp.courseOverview.statistics.presentationSco
             </div>
         </div>
         <div class="col-12 col-md-8">
-            <div class="row mt-3" *ngFor="let exerciseGroup of groupedExercises" id="{{ exerciseGroup.type }}-wrapper">
+            <div class="row mt-3" *ngFor="let exerciseGroup of ngxExerciseGroups" id="{{ exerciseGroup[0].type }}-wrapper">
                 <div class="col-12 mb-1 d-flex align-items-center">
-                    <div class="color-indicator" [style.background]="exerciseTitles[exerciseGroup.type].color"></div>
+                    <div class="color-indicator" [style.background]="exerciseTitles[exerciseGroup[0].type].color"></div>
                     <h2>
                         {{
                             'artemisApp.courseOverview.statistics.exerciseCount'
-                                | artemisTranslate: { number: exerciseGroup.names.length, type: exerciseTitles[exerciseGroup.type].name }
+                                | artemisTranslate: { number: exerciseGroup.length, type: exerciseTitles[exerciseGroup[0].type].name }
                         }}
                     </h2>
                 </div>
-                <div class="col-12 col-md-8">
-                    <canvas
-                        baseChart
-                        height="250"
-                        [datasets]="exerciseGroup.values"
-                        [labels]="exerciseGroup.names"
-                        [options]="barChartOptions"
-                        [legend]="false"
-                        [chartType]="barChartType"
-                    ></canvas>
+                <div #containerRef class="col-12 col-md-8">
+                    <ngx-charts-bar-horizontal-stacked
+                        [view]="[containerRef.offsetWidth, 250]"
+                        [scheme]="ngxBarColor"
+                        [results]="exerciseGroup"
+                        [xAxis]="true"
+                        [yAxis]="true"
+                        [xScaleMax]="exerciseGroup[0].xScaleMax"
+                        [barPadding]="exerciseGroup[0].barPadding"
+                    >
+                        <!-- This template is managing the tooltip content depending of the bar that is selected by the user and the properties of the corresponding exercise-->
+                        <ng-template #tooltipTemplate let-model="model">
+                            <div *ngIf="['Achieved (included)', 'Achieved bonus', 'Achieved (not included)', 'No due date'].includes(model.name)">
```

Comment:
Where do these strings come from? I feel like stuff like this should be stored in an enumeration (especially because you're re-using them in the lines below where they should be replaced as well then), but I don't know if there's a good reason for having raw strings here